CMakePresets.json

```diff
@@ -51,7 +51,7 @@
           "generator": "Ninja",
           "cacheVariables": {
               "VCPKG_TARGET_TRIPLET": "$env{VCPKG_TARGET_TRIPLET}",
-              "VCPKG_ROOT": "$env{VCPKG_ROOT}",
+              "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
```

Comment:
I left the presets to do everything in terms of `VCPKG_ROOT` rather than the toolchain file since we need it to figure out the compiler below. I'll try moving this to use `Z_VCPKG_ROOT_DIR` that gets set in the toolchain file.

Reply:
Never mind, that doesn't really work because `Z_VCPKG_ROOT_DIR` gets set by the toolchain file rather than supplied by the user which is required to be used in a preset.

Reply:
@2over12 @ekilmer any thoughts on making the cmake presets not manually specify a c/c++ compiler, and just go with whatever cmake finds by default?

Reply:
Is remill compatible with GCC now? If so, I think letting CMake find its own compiler is best

Reply:
I just tried with GCC and it seems that we're still using some things that are Clang-specific. In that case, would you prefer the preset to specify the compiler? That still seems weird to me in the debug build case where we would use a debug build of Clang.

Reply:
Thinking a bit more, I think **not** specifying the compiler is strictly better in this case. If you really want to record a specific compiler, then it should be in a [`CMakeUserPresets.json`](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html) file.

Caveat: If we really do want to test a specific compiler, it should be in a preset named something like "CI", so people are aware that the preset is for testing purposes and not generic use.

Reply:
Cool, I think we're basically in agreement that this can be removed from the preset.